docs: add project roadmap (#3448)
nuxt/framework: docs/content/4.community/4.roadmap.md

> # Project Roadmap

Reviewer: ```suggestion
# Roadmap
```

This is explicit enough :)

## 📢 Announcements

- [Nuxt 3 Beta](https://nuxtjs.org/announcements/nuxt3-beta/)
- [Nuxt 3 Release Candidate](https://github.com/nuxt/framework/discussions/3447)

## 📅 Release schedule

Release             | Expected date | Description
--------------------|---------------|---------------------------------------------------
`nuxt@3.0.0-rc.1`   | March 2022    | Nuxt v3 first release candidate
`nuxt@2.16`         | March 2022    | Nuxt v2 cumulative updates for future compatibility
`nuxt@3.0.0`        | June  2022    | Nuxt v3 stable release

### Current Releases

> Currently, [Nuxt 2](https://github.com/nuxt/nuxt.js) is considered **stable** version.
> Lastest updates on Nuxt 2.x are available via `nuxt-edge` npm package and expected to release alongside first release candidate of Nuxt 3 for future compatibility. [Nuxt 3](https://v3.nuxtjs.org) is **under development** and available as `nuxt3` npm package with automated releases for beta testing.

Reviewer 2: ```suggestion
Currently, [Nuxt 2](https://github.com/nuxt/nuxt.js) is considered the **stable** version.

The latest updates on Nuxt 2.x are available via the `nuxt-edge` npm package and are expected to be released alongside the first release candidate of Nuxt 3 for future compatibility. [Nuxt 3](https://v3.nuxtjs.org) is **under development** and available as the `nuxt3` npm package with automated releases for beta testing.
```

Release  | npm | Status      | Last Release | Docs |  Repository
---------|----|---------|--------------|------|-----------------
Nuxt 2.x | [`nuxt`](https://npmjs.com/package/nuxt)  | Maintenance | <a href="https://npmjs.com/package/nuxt"><img src="https://flat.badgen.net/npm/v/nuxt"></a>   | [2.x docs](https://nuxtjs.org/docs) | [nuxt/nuxt.js#2.x](https://github.com/nuxt/nuxt.js/tree/2.x)
Nuxt 2.x (edge*) | [`nuxt-edge`](https://npmjs.com/package/nuxt) | Maintenance | <a href="https://npmjs.com/package/nuxt-edge"><img src="https://flat.badgen.net/npm/v/nuxt-edge"></a>      | [2.x docs](https://nuxtjs.org/docs) | [nuxt/nuxt.js](https://github.com/nuxt/nuxt.js)
Nuxt 3.x (edge*) | [`nuxt3`](https://npmjs.com/package/nuxt3) | Development        | <a href="https://npmjs.com/package/nuxt3"><img src="https://flat.badgen.net/npm/v/nuxt3"></a>       | [3.x docs](https://v3.nuxtjs.org/) | [nuxt/framework](https://github.com/nuxt/framework)

(*) Edge releases are automated after each commit passing tests

## 🛣️ Roadmap

> In roadmap below, we listed major expected features that comming soon with Nuxt 3.

Reviewer 2: ```suggestion
In roadmap below are the major expected features that are coming soon with Nuxt 3.
```

--------------|---------------|--------|-----------------------
Nitropack  | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Stable and full-features server engine for Nuxt 3
Generate and Caching | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Route caching rules and full static generation
Test Utils   | Q1 2022 | [nuxt/framework#3198](https://github.com/nuxt/framework/issues/3198) | A rewrite of [nuxt/test-utils](https://github.com/nuxt/test-utils) for testing Nuxt 3 and new Modules
Nuxt 3 RC | Q1 2022 | [nuxt/framework#3447](https://github.com/nuxt/framework/discussions/3447) | Nuxt v3 first release candidate
Extending | Q1 2022 | [nuxt/framework#3222](https://github.com/nuxt/framework/issues/3222) | Native extending and theme support
Content v2 | Q1 2022     | -      | A rewrite of [nuxt/content](https://github.com/nuxt/content) for Nuxt 3 support
Devtools | Q2 2022       | -      | An integrated devtools experience for Nuxt
Auth     | Q2 2022       | -      | A rewrite of [nuxt-community/auth-module](https://github.com/nuxt-community/auth-module) for Nuxt 3 support

> Seo & PWA     | Q2 2022       | [nuxt/framework#1823](https://github.com/nuxt/framework/discussions/1823) | Migrating from [nuxt-community/pwa-module](https://github.com/nuxt-community/auth-module) for built-in SEO utils and Server-worker support

Reviewer 3: ```suggestion
SEO & PWA     | Q2 2022       | [nuxt/framework#1823](https://github.com/nuxt/framework/discussions/1823) | Migrating from [nuxt-community/pwa-module](https://github.com/nuxt-community/auth-module) for built-in SEO utils and service worker support
```

> Nitropack  | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Stable and full-features server engine for Nuxt 3
> Generate and Caching | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Route caching rules and full static generation
> Test Utils   | Q1 2022 | [nuxt/framework#3198](https://github.com/nuxt/framework/issues/3198) | A rewrite of [nuxt/test-utils](https://github.com/nuxt/test-utils) for testing Nuxt 3 and new Modules
> Nuxt 3 RC | Q1 2022 | [nuxt/framework#3447](https://github.com/nuxt/framework/discussions/3447) | Nuxt v3 first release candidate
> Extending | Q1 2022 | [nuxt/framework#3222](https://github.com/nuxt/framework/issues/3222) | Native extending and theme support
> Content v2 | Q1 2022     | -      | A rewrite of [nuxt/content](https://github.com/nuxt/content) for Nuxt 3 support
> Devtools | Q2 2022       | -      | An integrated devtools experience for Nuxt
> Auth     | Q2 2022       | -      | A rewrite of [nuxt-community/auth-module](https://github.com/nuxt-community/auth-module) for Nuxt 3 support
> Seo & PWA     | Q2 2022       | [nuxt/framework#1823](https://github.com/nuxt/framework/discussions/1823) | Migrating from [nuxt-community/pwa-module](https://github.com/nuxt-community/auth-module) for built-in SEO utils and Server-worker support
> Translations | Q2/Q3 2022 | [nuxt/translations#4](https://github.com/nuxt/translations/discussions/4) (private) | A collabrative project for stable translation process of Nuxt 3 docs. Currently pending for ideas and documentation tooling support (content v2 with remote sources).

Reviewer 2: ```suggestion
Nitropack  | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Stable and full-featured server engine for Nuxt 3
Generate and Caching | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Route caching rules and full static generation
Test Utils   | Q1 2022 | [nuxt/framework#3198](https://github.com/nuxt/framework/issues/3198) | A rewrite of [nuxt/test-utils](https://github.com/nuxt/test-utils) for testing Nuxt 3 and new modules
Nuxt 3 RC | Q1 2022 | [nuxt/framework#3447](https://github.com/nuxt/framework/discussions/3447) | Nuxt v3 first release candidate
Extending | Q1 2022 | [nuxt/framework#3222](https://github.com/nuxt/framework/issues/3222) | Native extending and theme support
Content v2 | Q1 2022     | -      | A rewrite of [nuxt/content](https://github.com/nuxt/content) for Nuxt 3 support
Devtools | Q2 2022       | -      | An integrated devtools experience for Nuxt
Auth     | Q2 2022       | -      | A rewrite of [nuxt-community/auth-module](https://github.com/nuxt-community/auth-module) for Nuxt 3 support
Seo & PWA     | Q2 2022       | [nuxt/framework#1823](https://github.com/nuxt/framework/discussions/1823) | Migrating from [nuxt-community/pwa-module](https://github.com/nuxt-community/auth-module) for built-in SEO utils and Server-worker support
Translations | Q2/Q3 2022 | [nuxt/translations#4](https://github.com/nuxt/translations/discussions/4) (private) | A collaborative project for a stable translation process for Nuxt 3 docs. Currently pending for ideas and documentation tooling support (content v2 with remote sources).
```

Reviewer: Seems I cannot commit this change @danielroe

## 📦 Core Modules

> Alongside with Nuxt Framework, there are modules that are vital for ecosystem. Their status will be updated below.

Reviewer 2: ```suggestion
In addition to the Nuxt framework, there are modules that are vital for the ecosystem. Their status will be updated below.
```

Nuxt 2.x (edge*) | [`nuxt-edge`](https://npmjs.com/package/nuxt) | Maintenance | <a href="https://npmjs.com/package/nuxt-edge"><img src="https://flat.badgen.net/npm/v/nuxt-edge"></a>      | [2.x docs](https://nuxtjs.org/docs) | [nuxt/nuxt.js](https://github.com/nuxt/nuxt.js)
Nuxt 3.x (edge*) | [`nuxt3`](https://npmjs.com/package/nuxt3) | Development        | <a href="https://npmjs.com/package/nuxt3"><img src="https://flat.badgen.net/npm/v/nuxt3"></a>       | [3.x docs](https://v3.nuxtjs.org/) | [nuxt/framework](https://github.com/nuxt/framework)

(*) Edge releases are automated after each commit passing tests

## 🛣️ Roadmap

In roadmap below are the major expected features that are coming soon with Nuxt 3.

💡 Check [Discussions](https://github.com/nuxt/framework/discussions) and [RFCs](https://github.com/nuxt/framework/discussions/categories/rfcs) for more upcoming features and ideas.

Milestone          | Expected date | Notes  | Description
--------------|---------------|--------|-----------------------

> Nitropack  | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Stable and full-features server engine for Nuxt 3
> Generate and Caching | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Route caching rules and full static generation
> Test Utils   | Q1 2022 | [nuxt/framework#3198](https://github.com/nuxt/framework/issues/3198) | A rewrite of [nuxt/test-utils](https://github.com/nuxt/test-utils) for testing Nuxt 3 and new Modules
> Nuxt 3 RC | Q1 2022 | [nuxt/framework#3447](https://github.com/nuxt/framework/discussions/3447) | Nuxt v3 first release candidate
> Extending | Q1 2022 | [nuxt/framework#3222](https://github.com/nuxt/framework/issues/3222) | Native extending and theme support
> Content v2 | Q1 2022     | -      | A rewrite of [nuxt/content](https://github.com/nuxt/content) for Nuxt 3 support
> Devtools | Q2 2022       | -      | An integrated devtools experience for Nuxt
> Auth     | Q2 2022       | -      | A rewrite of [nuxt-community/auth-module](https://github.com/nuxt-community/auth-module) for Nuxt 3 support
> SEO & PWA     | Q2 2022       | [nuxt/framework#1823](https://github.com/nuxt/framework/discussions/1823) | Migrating from [nuxt-community/pwa-module](https://github.com/nuxt-community/auth-module) for built-in SEO utils and service worker support
> Translations | Q2/Q3 2022 | [nuxt/translations#4](https://github.com/nuxt/translations/discussions/4) (private) | A collabrative project for stable translation process of Nuxt 3 docs. Currently pending for ideas and documentation tooling support (content v2 with remote sources).

Reviewer 2: ```suggestion
Nitropack  | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Stable and full-featured server engine for Nuxt 3
Generate and Caching | Q1 2022 | [nuxt/framework#3161](https://github.com/nuxt/framework/issues/3161) | Route caching rules and full static generation
Test Utils   | Q1 2022 | [nuxt/framework#3198](https://github.com/nuxt/framework/issues/3198) | A rewrite of [nuxt/test-utils](https://github.com/nuxt/test-utils) for testing Nuxt 3 and new modules
Nuxt 3 RC | Q1 2022 | [nuxt/framework#3447](https://github.com/nuxt/framework/discussions/3447) | Nuxt v3 first release candidate
Extending | Q1 2022 | [nuxt/framework#3222](https://github.com/nuxt/framework/issues/3222) | Native extending and theme support
Content v2 | Q1 2022     | -      | A rewrite of [nuxt/content](https://github.com/nuxt/content) for Nuxt 3 support
Devtools | Q2 2022       | -      | An integrated devtools experience for Nuxt
Auth     | Q2 2022       | -      | A rewrite of [nuxt-community/auth-module](https://github.com/nuxt-community/auth-module) for Nuxt 3 support
SEO & PWA     | Q2 2022       | [nuxt/framework#1823](https://github.com/nuxt/framework/discussions/1823) | Migrating from [nuxt-community/pwa-module](https://github.com/nuxt-community/auth-module) for built-in SEO utils and service worker support
Translations | Q2/Q3 2022 | [nuxt/translations#4](https://github.com/nuxt/translations/discussions/4) (private) | A collaborative project for a stable translation process for Nuxt 3 docs. Currently pending for ideas and documentation tooling support (content v2 with remote sources).
```

## 📦 Core Modules

In addition to the Nuxt framework, there are modules that are vital for the ecosystem. Their status will be updated below.

Module         | Status              | Nuxt Support | Repository | Description
---------------|---------------------|--------------|------------|-------------------

> Content 1.x    | Maintenance         | 2.x          | [nuxt/content](https://github.com/nuxt/content) | EOL

Reviewer 3: Question: Why is 1.x marked as "EOL" when there is not "alternative" yet? 🤔

Author: ```suggestion
Content 1.x    | Maintenance         | 2.x          | [nuxt/content](https://github.com/nuxt/content) | maintenance only
```
